src/modules/chat_fullscreen_popup/index.js

```diff
@@ -0,0 +1,102 @@
+import {ChatFullscreenPopupTypes, PlatformTypes, SettingIds} from '../../constants.js';
+import settings from '../../settings.js';
+import {loadModuleForPlatforms} from '../../utils/modules.js';
+import twitch from '../../utils/twitch.js';
+
+class ChatFullscreenPopup {
+  constructor() {
+    settings.on(`changed.${SettingIds.CHAT_FULLSCREEN_POPUP}`, () => this.togglePlugin());
+    this.updateElements();
+    this.fullscreenChatPopupShown = false;
+
+    window.addEventListener('fullscreenchange', () => this.fullscreenChanged());
+
+    this.fullscreenKeydownListener = this.fullscreenKeydown.bind(this);
+
+    // Keep track of keyup as well to prevent flickering when holding 'c'
+    this.keyupPressed = true;
+    this.fullscreenKeyupListener = this.fullscreenKeyup.bind(this);
+  }
+
+  togglePlugin() {
+    if (settings.get(SettingIds.CHAT_FULLSCREEN_POPUP) === ChatFullscreenPopupTypes.DISABLED) {
+      this.disableKeyListeners();
+      this.hideChatPopup();
+    }
+  }
+
+  updateElements() {
+    this.chatElement = document.querySelector('.stream-chat');
+    this.originalChatParentElement = this.chatElement.parentNode;
+  }
+
+  fullscreenChanged() {
+    if (
+      document.fullscreenElement &&
+      settings.get(SettingIds.CHAT_FULLSCREEN_POPUP) !== ChatFullscreenPopupTypes.DISABLED
```

Comment:
nit: `settings.get(SettingIds.CHAT_FULLSCREEN_POPUP) === ChatFullscreenPopupTypes.ENABLED`

but can be considered redundant if swapped to boolean